modules/RTC/RTCUtils.js

```diff
@@ -876,10 +876,34 @@ class RTCUtils extends Listenable {
                         { googSuspendBelowMinBitrate: true });
                 }
 
-                // There's no reason not to use this for p2p
-                this.p2pPcConstraints.optional.push({
-                    googSuspendBelowMinBitrate: true
-                });
+                /**
+                 * This option is used to enable the suspend video only for
+                 * part of the users on the P2P peer connection. The value of
+                 * the option is the ratio:
+                 * (users with suspended video enabled)/(all users).
+                 *
+                 * Note: The option is not documented because it is temporary
+                 * and only for internal testing purpose.
+                 *
+                 * @type {number}
+                 */
+                const forceP2PSuspendVideoRatio
+                    = options.testing.forceP2PSuspendVideoRatio;
```

Comment:
we can change this to
`const forceP2PSuspendVideoRatio = options.testing.forceP2PSuspendVideoRatio || 1`

and then get rid of the `typeof forceP2PSuspendVideoRatio !== 'number'` part of the if below

Reply:
Now we have only one place where we check `typeof forceP2PSuspendVideoRatio !== 'number' `. That's why I included it in the if condition. 

Also `const forceP2PSuspendVideoRatio = options.testing.forceP2PSuspendVideoRatio || 1` will handle only the case when the option is undefined. So it won't be the same.

I prefer to put the check in the if condition instead of assigning a value because this way we will save a Math.random call in the cases where the option has invalid type. It's not a big deal but in this case it is very easy to do and I don't think the code suffers from it.

Reply:
are you saying we need to handle the case where someone configures `forceP2PSuspendVideoRatio` to a value that isn't a number?  that doesn't seem like something we'd want to worry about...i doubt we do that for all the configuration values, do we?  even if we don't want to cover that, i still think it'd be better to handle it in the initialization of `forceP2PSuspendVideoRatio` rather than in the if.

Reply:
@bbaldino I think it's good practice to do the type checking. Personally for me it's hard to predict the result from `0.5 < false` or `0.5 < {}` :) I can't tell about the other config props. Do you prefer not to check for this? Do you think it's bad to check for this?

The benefit of not doing this in the initialization is that we won't execute Math.random for some cases and the code doesn't look uglier. I really don't see any benefits of handling this in the initialization. I'm sorry if I'm missing something! Why do you prefer to do this in the initialization?

Reply:
few thoughts on this:

1. type-checking can be great, but there are plenty of places where don't do it...why start doing it here?  do we do it consistently for all values in config?  if we want to start with this one, that's also ok...but i think it'd be better to do type-checking and setting of default-values for config variables in a central place rather than all over.

2. i definitely don't think we should do `0.5 < {}`, i was saying we should specifically avoid that by getting `forceP2PSuspendVideoRatio` to its expected type (a number between 0 and 1) as soon as possible.  in general, the earlier we do that, the more code below can safely assume that `forceP2PSuspendVideoRatio` will be a number between 0 and 1, and not have to do other error-checking (i realize it's only accessed in one place below now, but in the future that could change).

3. for someone else reading the code, i think the following if makes the intent much clearer:
```
    if (Math.random() < forceP2PSuspendVideoRatio)
```
plus, why do we consider a value that isn't a number to be equal to '100% probability'?

4. i don't think optimizing for avoiding a call to `Math.random()` is worth sacrificing (in my opinion) readability (without data to show it's causing performance problems).

Reply:
Thanks for the explanation!

1. I totally agree. But for now we are not doing this in the central place and we don't have a rule for it. So until we come up with something I think it is better for this feature to handle it where I use it instead of not handling it at all.
2. I believe your idea from 1) will address this, right?
3. Probably but if you compare:
```
const forceP2PSuspendVideoRatio = (typeof forceP2PSuspendVideoRatio !== 'number') ? 1 : forceP2PSuspendVideoRatio;
 if (Math.random() < forceP2PSuspendVideoRatio)
...
```
versus
```
 if (typeof forceP2PSuspendVideoRatio !== 'number' || Math.random() < forceP2PSuspendVideoRatio)
...
```

I definitely prefer the second one.

Also I consider invalid values as 100% probability because this is the default behavior and deployments that don't specify this parameter for example should continue working as they were working before. Does that make sense?

4. Sure! I agree, but as I explained in 3) I don't think we are sacrificing anything. I'd implement it your way before because we were using it in more than one place but now I definitely like the current version more. Also I don't believe that in future we'll use this parameter anywhere else because this is temporary code only for internal testing purpose that will probably be removed very soon(in the following weeks...).